crates/subspace-farmer/src/single_plot_farm.rs

```diff
@@ -834,13 +835,22 @@ impl<RC: RpcClient> OnSync for VerifyingPlotter<RC> {
         }
 
         let plotter = self.single_plot_plotter.clone();
+        let (farm_creation_start_sender, farm_creation_start_receiver) =
+            std::sync::mpsc::sync_channel(1);
+        // Makes sure background blocking task below is finished before this function's future is
+        // dropped
+        let _plot_pieces_guard = CallOnDrop::new(move || {
+            let _ = farm_creation_start_receiver.recv();
+        });
         tokio::task::spawn_blocking(move || {
-            plotter
-                .plot_pieces(PiecesToPlot {
-                    pieces,
-                    piece_indexes,
-                })
-                .map_err(Into::into)
+            plotter.plot_pieces(PiecesToPlot {
+                pieces,
+                piece_indexes,
+            })?;
+
+            let _ = farm_creation_start_sender.send(());
+
+            Ok(())
```

Comment:
Do we need this change, though? Await below should wait for all pieces to be plotted

Reply:
Because if await-er can be dropped any time without waiting for the future to complete and blocking task (thread) will continue running. What we want is to wait for that thread to finish before future is dropped.